add package: devilspie

rkitover opened the pull request:
This is a user daemon utility that reads configuration files in lisp for
matching X11 windows and performing actions on them.

For example, I have this file:

~/.devilspie/undecorate.ds

with this contents:

(if (or (is (window_class) "qterminal") (is (window_class)
"Xfce4-terminal"))
	(undecorate)
)

With the daemon launched from my xstartup, when a window matching the
class "qtermina" or "Xfce4-terminal" is opened, it will remove window
manager decorations from it giving me a borderless window without a
title bar.

Signed-off-by: Rafael Kitover <rkitover@gmail.com>

xeffyr commented on packages/devilspie/build.sh:
```diff
@@ -0,0 +1,16 @@
+TERMUX_PKG_SRCURL=http://deb.debian.org/debian/pool/main/d/devilspie/devilspie_${TERMUX_PKG_VERSION}.orig.tar.xz
+TERMUX_PKG_SHA256=2dd9d4b7424ac999f20c214b0ff71e8c5836d97003d7a8a7391105cd16d36deb
+TERMUX_PKG_DEPENDS="libx11, gettext"
+TERMUX_PKG_BUILD_DEPENDS="intltool"
```
Remove this. In Termux, `intltool` is unstable package, but repositories aren't shared. Also you can't use Termux build tools but only present in Docker image.

xeffyr commented on packages/devilspie/build.sh:
```diff
@@ -0,0 +1,16 @@
+	patch -p1 -i ${TERMUX_PKG_BUILDER_DIR}/fix_manpage_lintian_warnings.patch
+	patch -p1 -i ${TERMUX_PKG_BUILDER_DIR}/fix_memleak_in_my_wnck_get_viewport_start.patch
+	patch -p1 -i ${TERMUX_PKG_BUILDER_DIR}/fix_using_deprecated_wnck_functions.patch
+	patch -p1 -i ${TERMUX_PKG_BUILDER_DIR}/remove_unavailable_options_from_manpage.patch	
```
You can't apply patches in this way.

xeffyr replied:
https://github.com/termux/termux-packages/blob/master/build-package.sh#L374

https://github.com/termux/termux-packages/blob/master/scripts/build/termux_step_patch_package.sh#L1

xeffyr reviewed (changes requested):
* https://github.com/termux/termux-packages/wiki/Build-environment
* https://github.com/termux/termux-packages/wiki/Building-packages

Additionally you have lots of examples in ./packages directory.

rkitover committed:
add package: devilspie

This is a user daemon utility that reads configuration files in lisp for
matching X11 windows and performing actions on them.

For example, I have this file:

~/.devilspie/undecorate.ds

with this contents:

(if (or (is (window_class) "qterminal") (is (window_class)
"Xfce4-terminal"))
	(undecorate)
)

With the daemon launched from my xstartup, when a window matching the
class "qtermina" or "Xfce4-terminal" is opened, it will remove window
manager decorations from it giving me a borderless window without a
title bar.

Signed-off-by: Rafael Kitover <rkitover@gmail.com>
